### PR TITLE
(Android only) Force audio through speaker

### PR DESCRIPTION
**Pull request by arturzaitcev-tomtom**
- Created a function that returns an audio track with a preferred device set to speaker
- make sure that if the comparison fails, it is still set to the default device

### CODE REVIEW

**Comment by arturzaitcev-tomtom** [comment]
I think I should create a tag for this as well?

**Review by casparvanderheiden-tomtom — APPROVED** [review]

**Comment by casparvanderheiden-tomtom** [comment]
> I think I should create a tag for this as well?
Yes, so we can update the dependency in AmiGO

**Comment by arturzaitcev-tomtom** [comment]
I pushed the tag v1.3.3-amigo

File: android/src/main/java/net/no_mad/tts/TextToSpeechModule.java [hunk 1]
```diff
@@ -158,6 +156,7 @@ public void onBeginSynthesis(String utteranceId, int sampleRateInHz, int audioFo
                             tts.stop();
                         } else {
                             try {
+
```
Comment (johanvanderboom-tomtom) [inline android/src/main/java/net/no_mad/tts/TextToSpeechModule.java:159]: I think the call to `forceSpeakerRoute` may have been accidentally been removed with the latest changes?

Comment (arturzaitcev-tomtom) [reply android/src/main/java/net/no_mad/tts/TextToSpeechModule.java:159]: idk what is going on in my IDE.... let me fix that

**Review by johanvanderboom-tomtom — APPROVED** [review]

File: android/src/main/java/net/no_mad/tts/TextToSpeechModule.java [hunk 2]
```diff
@@ -893,4 +893,17 @@ private void writeAudioToFile(byte[] audio) {
             Log.e(TAG, "Failed to write audio to file e:" + e);
         }
     }
+
+    private void forceSpeakerRoute(AudioTrack audioTrack)
+    {
+        AudioManager audioManager = (AudioManager) context.getApplicationContext().getSystemService(Context.AUDIO_SERVICE);
```
Comment (johanvanderboom-tomtom) [inline android/src/main/java/net/no_mad/tts/TextToSpeechModule.java:899]: Note this does not compile, `context` is not available. `audioManager` should be used directly as it is setup during init phase. Also `AudioDeviceInfo` is not imported

Comment (arturzaitcev-tomtom) [reply android/src/main/java/net/no_mad/tts/TextToSpeechModule.java:899]: I will fix it

Comment (arturzaitcev-tomtom) [reply android/src/main/java/net/no_mad/tts/TextToSpeechModule.java:899]: I think I fixed my IDE issue now... so I can also see that

**Pull request merged**